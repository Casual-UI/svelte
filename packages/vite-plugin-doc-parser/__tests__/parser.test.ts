import { resolve } from 'node:path'
import process from 'node:process'
import { describe, expect, it } from 'vitest'
import { parse } from 'sveltedoc-parser'

describe('parser', async () => {
  it('CNotification.svelte', async () => {
    const api = await parse({
      filename: resolve(process.cwd(), '../ui/src/components/CNotification.svelte'),
      version: 3,
    })
    expect(api).toMatchSnapshot()
  })
})
