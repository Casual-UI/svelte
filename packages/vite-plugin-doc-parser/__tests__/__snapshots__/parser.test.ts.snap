// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parser > CNotification.svelte 1`] = `
{
  "components": [],
  "computed": [],
  "data": [],
  "description": null,
  "events": [],
  "keywords": [],
  "methods": [],
  "name": "CNotification",
  "refs": [],
  "slots": [
    {
      "description": "Customize the content of each notification item",
      "keywords": [
        {
          "description": "{any} itemInfo all the item's props will be passed to this slot. <br />For example: \`message\`, \`title\`, \`theme\`, \`timeout\`",
          "name": "param",
        },
      ],
      "name": "item",
      "parameters": [
        {
          "description": "all the item's props will be passed to this slot. <br />For example: \`message\`, \`title\`, \`theme\`, \`timeout\`",
          "name": "itemInfo",
          "optional": false,
          "type": {
            "kind": "type",
            "text": "any",
            "type": "any",
          },
        },
      ],
      "params": [
        {
          "description": "all the item's props will be passed to this slot. <br />For example: \`message\`, \`title\`, \`theme\`, \`timeout\`",
          "name": "itemInfo",
          "optional": false,
          "type": {
            "kind": "type",
            "text": "any",
            "type": "any",
          },
        },
      ],
      "visibility": "public",
    },
  ],
  "version": 3,
}
`;
